<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: Analyze Illumina's methylation arrays</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>
<h1> Analyze Illumina's methylation arrays
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;minfi&rsquo; version 1.18.6</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../doc/index.html">User guides, package vignettes and other documentation.</a></li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<p style="text-align: center;">
<a href="# "> </a>
<a href="#A">A</a>
<a href="#B">B</a>
<a href="#C">C</a>
<a href="#D">D</a>
<a href="#E">E</a>
<a href="#F">F</a>
<a href="#G">G</a>
<a href="#I">I</a>
<a href="#L">L</a>
<a href="#M">M</a>
<a href="#N">N</a>
<a href="#P">P</a>
<a href="#Q">Q</a>
<a href="#R">R</a>
<a href="#S">S</a>
</p>

<table width="100%">
<tr><td style="width: 25%;"><a href="minfi-package.html">minfi-package</a></td>
<td>Analyze Illumina's methylation arrays</td></tr>
</table>

<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="getQC.html">addQC</a></td>
<td>Estimate sample-specific quality control (QC) for methylation data</td></tr>
<tr><td style="width: 25%;"><a href="getSex.html">addSex</a></td>
<td>Estimating sample sex based on methylation data</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">addSnpInfo</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">annotation-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">annotation-method</a></td>
<td>GenomicRatioSet instances</td></tr>
</table>

<h2><a name="B">-- B --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="preprocessIllumina.html">bgcorrect.illumina</a></td>
<td>Perform preprocessing as Genome Studio.</td></tr>
<tr><td style="width: 25%;"><a href="blockFinder.html">blockFinder</a></td>
<td>Finds blocks of methylation differences for Illumina methylation arrays</td></tr>
<tr><td style="width: 25%;"><a href="bumphunter.html">bumphunter-method</a></td>
<td>Methods for function 'bumphunter' in Package 'minfi'</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="compartments.html">compartments</a></td>
<td>Estimates A/B compartments from Illumina methylation arrays</td></tr>
<tr><td style="width: 25%;"><a href="controlStripPlot.html">controlStripPlot</a></td>
<td>Plot control probe signals.</td></tr>
<tr><td style="width: 25%;"><a href="cpgCollapse.html">cpgCollapse</a></td>
<td>Collapse methylation values of adjacent CpGs into a summary value.</td></tr>
<tr><td style="width: 25%;"><a href="compartments.html">createCorMatrix</a></td>
<td>Estimates A/B compartments from Illumina methylation arrays</td></tr>
</table>

<h2><a name="D">-- D --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="densityBeanPlot.html">densityBeanPlot</a></td>
<td>Density bean plots of methylation Beta values.</td></tr>
<tr><td style="width: 25%;"><a href="densityPlot.html">densityPlot</a></td>
<td>Density plots of methylation Beta values.</td></tr>
<tr><td style="width: 25%;"><a href="detectionP.html">detectionP</a></td>
<td>Detection p-values for all probed genomic positions.</td></tr>
<tr><td style="width: 25%;"><a href="dmpFinder.html">dmpFinder</a></td>
<td>Find differentially methylated positions</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">dropLociWithSnps</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">dropMethylationLoci</a></td>
<td>MethylSet instances</td></tr>
</table>

<h2><a name="E">-- E --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="estimateCellCounts.html">estimateCellCounts</a></td>
<td>Cell Proportion Estimation</td></tr>
<tr><td style="width: 25%;"><a href="compartments.html">extractAB</a></td>
<td>Estimates A/B compartments from Illumina methylation arrays</td></tr>
</table>

<h2><a name="F">-- F --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">featureNames-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">featureNames-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="fixMethOutliers.html">fixMethOutliers</a></td>
<td>Fix methylation outliers</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="gaphunter.html">gaphunter</a></td>
<td>Find gap signals in 450k data</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">GenomicMethylSet</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">GenomicMethylSet-class</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">GenomicRatioSet</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">GenomicRatioSet-class</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">getAnnotation</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">getAnnotationObject</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getBeta</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">getBeta-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">getBeta-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getBeta-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">getBeta-method</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">getBeta-method</a></td>
<td>RatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">getCN</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">getCN-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">getCN-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getCN-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">getCN-method</a></td>
<td>RatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">getControlAddress</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">getControlTypes</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="getGenomicRatioSetFromGEO.html">getGenomicRatioSetFromGEO</a></td>
<td>Reading Illumina methylation array data from GEO.</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">getGreen</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">getIslandStatus</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">getLocations</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getM</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">getM-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">getM-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getM-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">getM-method</a></td>
<td>RatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">getManifest</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationAnnotation-class.html">getManifest-method</a></td>
<td>Class 'IlluminaMethylationAnnotation'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">getManifest-method</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getManifest-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">getManifest-method</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">getManifestInfo</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getMeth</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">getMeth-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getMeth-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="utils.html">getMethSignal</a></td>
<td>Various utilities</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">getOOB</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">getProbeInfo</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">getProbeType</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="getQC.html">getQC</a></td>
<td>Estimate sample-specific quality control (QC) for methylation data</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">getRed</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="getSex.html">getSex</a></td>
<td>Estimating sample sex based on methylation data</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">getSnpBeta</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="getAnnotation.html">getSnpInfo</a></td>
<td>Accessing annotation for Illumina methylation objects</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getUnmeth</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">getUnmeth-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">getUnmeth-method</a></td>
<td>MethylSet instances</td></tr>
</table>

<h2><a name="I">-- I --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="IlluminaMethylationAnnotation-class.html">IlluminaMethylationAnnotation</a></td>
<td>Class 'IlluminaMethylationAnnotation'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationAnnotation-class.html">IlluminaMethylationAnnotation-class</a></td>
<td>Class 'IlluminaMethylationAnnotation'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">IlluminaMethylationManifest</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">IlluminaMethylationManifest-class</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="logit2.html">ilogit2</a></td>
<td>logit in base 2.</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">initialize-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">initialize-method</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">initialize-method</a></td>
<td>RatioSet instances</td></tr>
</table>

<h2><a name="L">-- L --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="logit2.html">logit2</a></td>
<td>logit in base 2.</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="makeGenomicRatioSetFromMatrix.html">makeGenomicRatioSetFromMatrix</a></td>
<td>Make a GenomicRatioSet from a matrix</td></tr>
<tr><td style="width: 25%;"><a href="mapToGenome-methods.html">mapToGenome</a></td>
<td>Mapping methylation data to the genome</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">mapToGenome-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">mapToGenome-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="mapToGenome-methods.html">mapToGenome-method</a></td>
<td>Mapping methylation data to the genome</td></tr>
<tr><td style="width: 25%;"><a href="mapToGenome-methods.html">mapToGenome-methods</a></td>
<td>Mapping methylation data to the genome</td></tr>
<tr><td style="width: 25%;"><a href="mdsPlot.html">mdsPlot</a></td>
<td>Multi-dimensional scaling plots giving an overview of similarities and differences between samples.</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">MethylSet</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">MethylSet-class</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="minfi-package.html">minfi</a></td>
<td>Analyze Illumina's methylation arrays</td></tr>
<tr><td style="width: 25%;"><a href="minfi-deprecated.html">minfi-deprecated</a></td>
<td>Deprecated functions in package 'minfi'</td></tr>
<tr><td style="width: 25%;"><a href="minfiQC.html">minfiQC</a></td>
<td>easy one-step QC of methylation object</td></tr>
</table>

<h2><a name="N">-- N --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="preprocessIllumina.html">normalize.illumina.control</a></td>
<td>Perform preprocessing as Genome Studio.</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">pData-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">pData-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="plotBetasByType.html">plotBetasByType</a></td>
<td>Plot the overall distribution of beta values and the distributions of the Infinium I and II probe types.</td></tr>
<tr><td style="width: 25%;"><a href="plotCpg.html">plotCpg</a></td>
<td>Plot methylation values at an single genomic position</td></tr>
<tr><td style="width: 25%;"><a href="getQC.html">plotQC</a></td>
<td>Estimate sample-specific quality control (QC) for methylation data</td></tr>
<tr><td style="width: 25%;"><a href="getSex.html">plotSex</a></td>
<td>Estimating sample sex based on methylation data</td></tr>
<tr><td style="width: 25%;"><a href="preprocessFunnorm.html">preprocessFunnorm</a></td>
<td>Functional normalization for Illumina 450k arrays</td></tr>
<tr><td style="width: 25%;"><a href="preprocessIllumina.html">preprocessIllumina</a></td>
<td>Perform preprocessing as Genome Studio.</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">preprocessMethod</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">preprocessMethod-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">preprocessMethod-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">preprocessMethod-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">preprocessMethod-method</a></td>
<td>RatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="preprocessNoob.html">preprocessNoob</a></td>
<td>The Noob background correction for 450k arrays.</td></tr>
<tr><td style="width: 25%;"><a href="preprocessQuantile.html">preprocessQuantile</a></td>
<td>Stratified quantile normalization for an Illumina methylation array.</td></tr>
<tr><td style="width: 25%;"><a href="preprocessRaw.html">preprocessRaw</a></td>
<td>Creation of a MethylSet without normalization</td></tr>
<tr><td style="width: 25%;"><a href="preprocessSwan.html">preprocessSWAN</a></td>
<td>Subset-quantile Within Array Normalisation for Illumina Infinium HumanMethylation450 BeadChips</td></tr>
</table>

<h2><a name="Q">-- Q --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="qcReport.html">qcReport</a></td>
<td>QC report for Illumina Infinium Human Methylation 450k arrays</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="ratioConvert-methods.html">ratioConvert</a></td>
<td>Converting methylation signals to ratios (Beta or M-values)</td></tr>
<tr><td style="width: 25%;"><a href="ratioConvert-methods.html">ratioConvert-method</a></td>
<td>Converting methylation signals to ratios (Beta or M-values)</td></tr>
<tr><td style="width: 25%;"><a href="ratioConvert-methods.html">ratioConvert-methods</a></td>
<td>Converting methylation signals to ratios (Beta or M-values)</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">RatioSet</a></td>
<td>RatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">RatioSet-class</a></td>
<td>RatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="read.450k.html">read.450k</a></td>
<td>Parsing IDAT files from Illumina methylation arrays.</td></tr>
<tr><td style="width: 25%;"><a href="read.450k.exp.html">read.450k.exp</a></td>
<td>Reads an entire 450k experiment using a sample sheet</td></tr>
<tr><td style="width: 25%;"><a href="read.450k.sheet.html">read.450k.sheet</a></td>
<td>Reading an Illumina methylation sample sheet</td></tr>
<tr><td style="width: 25%;"><a href="read.metharray.html">read.metharray</a></td>
<td>Parsing IDAT files from Illumina methylation arrays.</td></tr>
<tr><td style="width: 25%;"><a href="read.metharray.exp.html">read.metharray.exp</a></td>
<td>Reads an entire metharray experiment using a sample sheet</td></tr>
<tr><td style="width: 25%;"><a href="read.metharray.sheet.html">read.metharray.sheet</a></td>
<td>Reading an Illumina methylation sample sheet</td></tr>
<tr><td style="width: 25%;"><a href="readGEORawFile.html">readGEORawFile</a></td>
<td>Read in Unmethylated and Methylated signals from a GEO raw file.</td></tr>
<tr><td style="width: 25%;"><a href="readTCGA.html">readTCGA</a></td>
<td>Read in tab deliminited file in the TCGA format</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">RGChannelSet</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">RGChannelSet-class</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">RGChannelSetExtended</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">RGChannelSetExtended-class</a></td>
<td>Class '"RGChannelSet"'</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">sampleNames-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">sampleNames-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicMethylSet-class.html">show-method</a></td>
<td>GenomicMethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="GenomicRatioSet-class.html">show-method</a></td>
<td>GenomicRatioSet instances</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationAnnotation-class.html">show-method</a></td>
<td>Class 'IlluminaMethylationAnnotation'</td></tr>
<tr><td style="width: 25%;"><a href="IlluminaMethylationManifest-class.html">show-method</a></td>
<td>Class '"IlluminaMethylationManifest"'</td></tr>
<tr><td style="width: 25%;"><a href="MethylSet-class.html">show-method</a></td>
<td>MethylSet instances</td></tr>
<tr><td style="width: 25%;"><a href="RGChannelSet-class.html">show-method</a></td>
<td>Class '"RGChannelSet"'</td></tr>
<tr><td style="width: 25%;"><a href="RatioSet-class.html">show-method</a></td>
<td>RatioSet instances</td></tr>
</table>
</body></html>
