<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head><title>R: Differential DNA loop calling from ChIA-PET data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>
<h1> Differential DNA loop calling from ChIA-PET data
<img class="toplogo" src="../../../doc/html/Rlogo.svg" alt="[R logo]" />
</h1>
<hr/>
<div style="text-align: center;">
<a href="../../../doc/html/packages.html"><img class="arrow" src="../../../doc/html/left.jpg" alt="[Up]" /></a>
<a href="../../../doc/html/index.html"><img class="arrow" src="../../../doc/html/up.jpg" alt="[Top]" /></a>
</div><h2>Documentation for package &lsquo;diffloop&rsquo; version 1.1.2</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
<li><a href="../NEWS">Package NEWS</a>.</li>
</ul>

<h2>Help Pages</h2>


<p style="text-align: center;">
<a href="# "> </a>
<a href="#A">A</a>
<a href="#B">B</a>
<a href="#C">C</a>
<a href="#D">D</a>
<a href="#F">F</a>
<a href="#G">G</a>
<a href="#H">H</a>
<a href="#I">I</a>
<a href="#K">K</a>
<a href="#L">L</a>
<a href="#M">M</a>
<a href="#N">N</a>
<a href="#P">P</a>
<a href="#Q">Q</a>
<a href="#R">R</a>
<a href="#S">S</a>
<a href="#T">T</a>
<a href="#U">U</a>
<a href="#misc">misc</a>
</p>

<table width="100%">
<tr><td style="width: 25%;"><a href="diffloop.html">diffloop-package</a></td>
<td>diffloop: A package for differential DNA loop calling from ChIA-PET data</td></tr>
</table>

<h2><a name="A">-- A --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="addchr.html">addchr</a></td>
<td>Add 'chr' to GRanges seqnames</td></tr>
<tr><td style="width: 25%;"><a href="addchr.html">addchr-method</a></td>
<td>Add 'chr' to GRanges seqnames</td></tr>
<tr><td style="width: 25%;"><a href="annotateAnchors.html">annotateAnchors</a></td>
<td>Add meta data column to anchors based on GRanges</td></tr>
<tr><td style="width: 25%;"><a href="annotateAnchors.html">annotateAnchors-method</a></td>
<td>Add meta data column to anchors based on GRanges</td></tr>
<tr><td style="width: 25%;"><a href="annotateAnchors.bed.html">annotateAnchors.bed</a></td>
<td>Add meta data column to anchors based on bedgraph scores</td></tr>
<tr><td style="width: 25%;"><a href="annotateAnchors.bed.html">annotateAnchors.bed-method</a></td>
<td>Add meta data column to anchors based on bedgraph scores</td></tr>
<tr><td style="width: 25%;"><a href="annotateAnchors.bigwig.html">annotateAnchors.bigwig</a></td>
<td>Add meta data column to anchors based on .bigwig</td></tr>
<tr><td style="width: 25%;"><a href="annotateAnchors.bigwig.html">annotateAnchors.bigwig-method</a></td>
<td>Add meta data column to anchors based on .bigwig</td></tr>
<tr><td style="width: 25%;"><a href="annotateLoops.html">annotateLoops</a></td>
<td>Annotate loops as Enhancer-Promoter or CTCF-CTCF</td></tr>
<tr><td style="width: 25%;"><a href="annotateLoops.html">annotateLoops-method</a></td>
<td>Annotate loops as Enhancer-Promoter or CTCF-CTCF</td></tr>
<tr><td style="width: 25%;"><a href="annotateLoops.dge.html">annotateLoops.dge</a></td>
<td>Annotate enhancer-promoter loops with differential gene expression</td></tr>
<tr><td style="width: 25%;"><a href="annotateLoops.dge.html">annotateLoops.dge-method</a></td>
<td>Annotate enhancer-promoter loops with differential gene expression</td></tr>
</table>

<h2><a name="B">-- B --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="bedToGRanges.html">bedToGRanges</a></td>
<td>Read a file and make a GRanges object</td></tr>
<tr><td style="width: 25%;"><a href="bedToGRanges.html">bedToGRanges-method</a></td>
<td>Read a file and make a GRanges object</td></tr>
</table>

<h2><a name="C">-- C --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="calcLDSizeFactors.html">calcLDSizeFactors</a></td>
<td>Compute normalizing factors for each sample</td></tr>
<tr><td style="width: 25%;"><a href="calcLDSizeFactors.html">calcLDSizeFactors-method</a></td>
<td>Compute normalizing factors for each sample</td></tr>
<tr><td style="width: 25%;"><a href="callCCDs.html">callCCDs</a></td>
<td>Compute Chromatin Contact Domains (CCDs)</td></tr>
<tr><td style="width: 25%;"><a href="callCCDs.html">callCCDs-method</a></td>
<td>Compute Chromatin Contact Domains (CCDs)</td></tr>
<tr><td style="width: 25%;"><a href="computeBoundaryScores.html">computeBoundaryScores</a></td>
<td>Compute boundary scores for genomic loci in between anchors</td></tr>
<tr><td style="width: 25%;"><a href="computeBoundaryScores.html">computeBoundaryScores-method</a></td>
<td>Compute boundary scores for genomic loci in between anchors</td></tr>
</table>

<h2><a name="D">-- D --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="diffloop.html">diffloop</a></td>
<td>diffloop: A package for differential DNA loop calling from ChIA-PET data</td></tr>
<tr><td style="width: 25%;"><a href="dim-loops-method.html">dim-method</a></td>
<td>See dimensions of loops object</td></tr>
</table>

<h2><a name="F">-- F --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="featureTest.html">featureTest</a></td>
<td>Combined association test for all loops in a defined region</td></tr>
<tr><td style="width: 25%;"><a href="featureTest.html">featureTest-method</a></td>
<td>Combined association test for all loops in a defined region</td></tr>
<tr><td style="width: 25%;"><a href="filterLoops.html">filterLoops</a></td>
<td>Filter loops</td></tr>
<tr><td style="width: 25%;"><a href="filterLoops.html">filterLoops-method</a></td>
<td>Filter loops</td></tr>
<tr><td style="width: 25%;"><a href="filterSpanningLoops.html">filterSpanningLoops</a></td>
<td>Retain loops spanning some genomic feature</td></tr>
<tr><td style="width: 25%;"><a href="filterSpanningLoops.html">filterSpanningLoops-method</a></td>
<td>Retain loops spanning some genomic feature</td></tr>
</table>

<h2><a name="G">-- G --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="geneinfo.html">geneinfo</a></td>
<td>Human/mouse exon locations</td></tr>
<tr><td style="width: 25%;"><a href="getHumanGenes.html">getHumanGenes</a></td>
<td>Get protein coding gene regions</td></tr>
<tr><td style="width: 25%;"><a href="getHumanGenes.html">getHumanGenes-method</a></td>
<td>Get protein coding gene regions</td></tr>
<tr><td style="width: 25%;"><a href="getHumanTSS.html">getHumanTSS</a></td>
<td>Get Human Transcription Start Sites</td></tr>
<tr><td style="width: 25%;"><a href="getHumanTSS.html">getHumanTSS-method</a></td>
<td>Get Human Transcription Start Sites</td></tr>
<tr><td style="width: 25%;"><a href="getMouseGenes.html">getMouseGenes</a></td>
<td>Get protein coding gene regions</td></tr>
<tr><td style="width: 25%;"><a href="getMouseGenes.html">getMouseGenes-method</a></td>
<td>Get protein coding gene regions</td></tr>
<tr><td style="width: 25%;"><a href="getMouseTSS.html">getMouseTSS</a></td>
<td>Get Mouse Transcription Start Sites</td></tr>
<tr><td style="width: 25%;"><a href="getMouseTSS.html">getMouseTSS-method</a></td>
<td>Get Mouse Transcription Start Sites</td></tr>
</table>

<h2><a name="H">-- H --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="head-loops-method.html">head-method</a></td>
<td>Extract first part of loops object</td></tr>
<tr><td style="width: 25%;"><a href="human.genes.html">human.genes</a></td>
<td>Human protein coding genes</td></tr>
</table>

<h2><a name="I">-- I --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="interchromosomal.html">interchromosomal</a></td>
<td>Loops between chromosomes</td></tr>
<tr><td style="width: 25%;"><a href="interchromosomal.html">interchromosomal-method</a></td>
<td>Loops between chromosomes</td></tr>
<tr><td style="width: 25%;"><a href="intrachromosomal.html">intrachromosomal</a></td>
<td>Loops within chromosomes</td></tr>
<tr><td style="width: 25%;"><a href="intrachromosomal.html">intrachromosomal-method</a></td>
<td>Loops within chromosomes</td></tr>
</table>

<h2><a name="K">-- K --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="keepCTCFloops.html">keepCTCFloops</a></td>
<td>Keep CTCF loops</td></tr>
<tr><td style="width: 25%;"><a href="keepCTCFloops.html">keepCTCFloops-method</a></td>
<td>Keep CTCF loops</td></tr>
<tr><td style="width: 25%;"><a href="keepEPloops.html">keepEPloops</a></td>
<td>Keep enhancer-promoter loops</td></tr>
<tr><td style="width: 25%;"><a href="keepEPloops.html">keepEPloops-method</a></td>
<td>Keep enhancer-promoter loops</td></tr>
</table>

<h2><a name="L">-- L --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="loopFit.html">loopFit</a></td>
<td>Fit model for association testing</td></tr>
<tr><td style="width: 25%;"><a href="loopfit-class.html">loopfit</a></td>
<td>A class to represent ChIA-PET interaction data and an edgeR fit.</td></tr>
<tr><td style="width: 25%;"><a href="loopfit-class.html">loopfit-class</a></td>
<td>A class to represent ChIA-PET interaction data and an edgeR fit.</td></tr>
<tr><td style="width: 25%;"><a href="loopFit.html">loopFit-method</a></td>
<td>Fit model for association testing</td></tr>
<tr><td style="width: 25%;"><a href="loopGenes.html">loopGenes</a></td>
<td>Determine genes contained within loops</td></tr>
<tr><td style="width: 25%;"><a href="loopGenes.html">loopGenes-method</a></td>
<td>Determine genes contained within loops</td></tr>
<tr><td style="width: 25%;"><a href="loopMetrics.html">loopMetrics</a></td>
<td>Types of loops</td></tr>
<tr><td style="width: 25%;"><a href="loopMetrics.html">loopMetrics-method</a></td>
<td>Types of loops</td></tr>
<tr><td style="width: 25%;"><a href="loopPlot.html">loopPlot</a></td>
<td>Visualize looping</td></tr>
<tr><td style="width: 25%;"><a href="loopPlot.html">loopPlot-method</a></td>
<td>Visualize looping</td></tr>
<tr><td style="width: 25%;"><a href="loops-class.html">loops</a></td>
<td>A class to represent ChIA-PET interaction data and annotations</td></tr>
<tr><td style="width: 25%;"><a href="loops-class.html">loops-class</a></td>
<td>A class to represent ChIA-PET interaction data and annotations</td></tr>
<tr><td style="width: 25%;"><a href="loops.small.html">loops.small</a></td>
<td>chr1:36000000-36300000 loops</td></tr>
<tr><td style="width: 25%;"><a href="loopsMake.html">loopsMake</a></td>
<td>Read preprocessed ChIA-PET data from dnaloop</td></tr>
<tr><td style="width: 25%;"><a href="loopsMake.html">loopsMake-method</a></td>
<td>Read preprocessed ChIA-PET data from dnaloop</td></tr>
<tr><td style="width: 25%;"><a href="loopsMake.mango.html">loopsMake.mango</a></td>
<td>Read preprocessed ChIA-PET data from mango</td></tr>
<tr><td style="width: 25%;"><a href="loopsMake.mango.html">loopsMake.mango-method</a></td>
<td>Read preprocessed ChIA-PET data from mango</td></tr>
<tr><td style="width: 25%;"><a href="loopsSubset.html">loopsSubset</a></td>
<td>Subset two difloop objects</td></tr>
<tr><td style="width: 25%;"><a href="loopsSubset.html">loopsSubset-method</a></td>
<td>Subset two difloop objects</td></tr>
<tr><td style="width: 25%;"><a href="loopTest.html">loopTest</a></td>
<td>Differential Loop Calling</td></tr>
<tr><td style="width: 25%;"><a href="loopTest.html">loopTest-method</a></td>
<td>Differential Loop Calling</td></tr>
<tr><td style="width: 25%;"><a href="loopWidth.html">loopWidth</a></td>
<td>Loop widths</td></tr>
<tr><td style="width: 25%;"><a href="loopWidth.html">loopWidth-method</a></td>
<td>Loop widths</td></tr>
</table>

<h2><a name="M">-- M --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="mangoCorrection.html">mangoCorrection</a></td>
<td>Perform mango bias correction</td></tr>
<tr><td style="width: 25%;"><a href="mangoCorrection.html">mangoCorrection-method</a></td>
<td>Perform mango bias correction</td></tr>
<tr><td style="width: 25%;"><a href="manyLoopPlots.html">manyLoopPlots</a></td>
<td>Plot several loop regions</td></tr>
<tr><td style="width: 25%;"><a href="manyLoopPlots.html">manyLoopPlots-method</a></td>
<td>Plot several loop regions</td></tr>
<tr><td style="width: 25%;"><a href="mergeAnchors.html">mergeAnchors</a></td>
<td>Combine nearby anchors into one peak</td></tr>
<tr><td style="width: 25%;"><a href="mergeAnchors.html">mergeAnchors-method</a></td>
<td>Combine nearby anchors into one peak</td></tr>
</table>

<h2><a name="N">-- N --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="numAnchors.html">numAnchors</a></td>
<td>Get number of anchors in each sample</td></tr>
<tr><td style="width: 25%;"><a href="numAnchors.html">numAnchors-method</a></td>
<td>Get number of anchors in each sample</td></tr>
<tr><td style="width: 25%;"><a href="numLoops.html">numLoops</a></td>
<td>Per-sample loop quantities</td></tr>
<tr><td style="width: 25%;"><a href="numLoops.html">numLoops-method</a></td>
<td>Per-sample loop quantities</td></tr>
</table>

<h2><a name="P">-- P --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="padGRanges.html">padGRanges</a></td>
<td>Pad a GRanges object</td></tr>
<tr><td style="width: 25%;"><a href="padGRanges.html">padGRanges-method</a></td>
<td>Pad a GRanges object</td></tr>
<tr><td style="width: 25%;"><a href="pcaPlot.html">pcaPlot</a></td>
<td>Visualize sample relationships</td></tr>
<tr><td style="width: 25%;"><a href="pcaPlot.html">pcaPlot-method</a></td>
<td>Visualize sample relationships</td></tr>
<tr><td style="width: 25%;"><a href="plotTopLoops.html">plotTopLoops</a></td>
<td>Plot the most significant loops</td></tr>
<tr><td style="width: 25%;"><a href="plotTopLoops.html">plotTopLoops-method</a></td>
<td>Plot the most significant loops</td></tr>
</table>

<h2><a name="Q">-- Q --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="quickAssoc.html">quickAssoc</a></td>
<td>Perform quick differential loop calling</td></tr>
<tr><td style="width: 25%;"><a href="quickAssoc.html">quickAssoc-method</a></td>
<td>Perform quick differential loop calling</td></tr>
<tr><td style="width: 25%;"><a href="quickAssocVoom.html">quickAssocVoom</a></td>
<td>Perform quick differential loop calling</td></tr>
<tr><td style="width: 25%;"><a href="quickAssocVoom.html">quickAssocVoom-method</a></td>
<td>Perform quick differential loop calling</td></tr>
</table>

<h2><a name="R">-- R --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="removeRegion.html">removeRegion</a></td>
<td>Remove region from loops object</td></tr>
<tr><td style="width: 25%;"><a href="removeRegion.html">removeRegion-method</a></td>
<td>Remove region from loops object</td></tr>
<tr><td style="width: 25%;"><a href="removeSelfLoops.html">removeSelfLoops</a></td>
<td>Remove self loops</td></tr>
<tr><td style="width: 25%;"><a href="removeSelfLoops.html">removeSelfLoops-method</a></td>
<td>Remove self loops</td></tr>
<tr><td style="width: 25%;"><a href="rmchr.html">rmchr</a></td>
<td>Remove 'chr' from GRanges seqnames</td></tr>
<tr><td style="width: 25%;"><a href="rmchr.html">rmchr-method</a></td>
<td>Remove 'chr' from GRanges seqnames</td></tr>
</table>

<h2><a name="S">-- S --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="sampleNames-loops-method.html">sampleNames-method</a></td>
<td>Grab/Update Sample Names</td></tr>
<tr><td style="width: 25%;"><a href="sampleNames-loops-method.html">sampleNames&lt;--method</a></td>
<td>Grab/Update Sample Names</td></tr>
<tr><td style="width: 25%;"><a href="slidingWindowTest.html">slidingWindowTest</a></td>
<td>Combined association test for all loops in a defined region</td></tr>
<tr><td style="width: 25%;"><a href="slidingWindowTest.html">slidingWindowTest-method</a></td>
<td>Combined association test for all loops in a defined region</td></tr>
<tr><td style="width: 25%;"><a href="splitSamples.html">splitSamples</a></td>
<td>Split samples into their own loops object</td></tr>
<tr><td style="width: 25%;"><a href="splitSamples.html">splitSamples-method</a></td>
<td>Split samples into their own loops object</td></tr>
<tr><td style="width: 25%;"><a href="subsetLoops.html">subsetLoops</a></td>
<td>Subset loops</td></tr>
<tr><td style="width: 25%;"><a href="subsetLoops.html">subsetLoops-method</a></td>
<td>Subset loops</td></tr>
<tr><td style="width: 25%;"><a href="subsetRegion.html">subsetRegion</a></td>
<td>Extract region from loops object</td></tr>
<tr><td style="width: 25%;"><a href="subsetRegion.html">subsetRegion-method</a></td>
<td>Extract region from loops object</td></tr>
<tr><td style="width: 25%;"><a href="subsetRegionAB.html">subsetRegionAB</a></td>
<td>Retain loops that have anchors in two specified regions</td></tr>
<tr><td style="width: 25%;"><a href="subsetRegionAB.html">subsetRegionAB-method</a></td>
<td>Retain loops that have anchors in two specified regions</td></tr>
<tr><td style="width: 25%;"><a href="summary-loops-method.html">summary-method</a></td>
<td>Link the anchors and interactions back together</td></tr>
</table>

<h2><a name="T">-- T --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="tail-loops-method.html">tail-method</a></td>
<td>Extract last part of loops object</td></tr>
<tr><td style="width: 25%;"><a href="topLoops.html">topLoops</a></td>
<td>Grab top loops</td></tr>
<tr><td style="width: 25%;"><a href="topLoops.html">topLoops-method</a></td>
<td>Grab top loops</td></tr>
</table>

<h2><a name="U">-- U --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="union-loops-loops-method.html">union-method</a></td>
<td>Combine two loops objects</td></tr>
<tr><td style="width: 25%;"><a href="updateLDGroups.html">updateLDGroups</a></td>
<td>Update groups in colData for loops object</td></tr>
<tr><td style="width: 25%;"><a href="updateLDGroups.html">updateLDGroups-method</a></td>
<td>Update groups in colData for loops object</td></tr>
</table>

<h2><a name="misc">-- misc --</a></h2>

<table width="100%">
<tr><td style="width: 25%;"><a href="sub-loops-numeric-numeric-missing-method.html">[-method</a></td>
<td>Extract parts of a loops object</td></tr>
</table>
</body></html>
