---
graphics: yes
---
# **DNAlandscapeR Guide**
<br>
#### **Contents:**
* [Advanced plotting options](#apo)
* [Interactive zooming](#iz)
* [Visualizing Local Data](#uld)
* [Creating .rds for ChIA-PET Visualization](#crL)
* [Creating .rds for HiC Visualization](#hic)
* [FAQ](#faq)
<hr>

<a name="apo"></a>
### **Advanced plotting options**
Though listed in the main sidebar, the **Genome Annotation** drop-down box supports a basic gene body annotation as well as more advanced track. In the default gene bodies track, only protein-coding genes whose entire transcription body is present in the specified region will be displayed. The second track, "Detailed Gene Annotation", displays exon-level resolution of gene bodies that are not necessarily protein coding genes. Additionally, genes only partially present in the region are displayed. Other features in the **Advanced Options** box include:
* **log2 Transform .bigwig files--** When this box is checked, the scores associated with epigenetic peaks will be log2 transformed. This only applies to ReadDepth tracks (e.g. RNA-Seq, DNase, ChIP-Seq, etc.) and does NOT apply to methylation.
* **Display single anchors--** **DNAlandscapeR** only plots loops that have both anchors in the specified region. When this option is selected, small green bumps will be displayed in areas where one side of a loop anchors in the region. Consider zooming out to see the full loop.
* **Plot gene region--** This option allows users to display a region based on gene name where the entire gene body will be displayed as well as a buffer region on either side of the gene body. 
* **Displayed genes--** As large landscapes can have many genes displayed, this configuration enables users to remove genes that they do not wish to be displayed on the gene annotation track. 
Finally, we note one other configuration option on the sidebar that allows users to upload a regions .bed file with the "<<<" and ">>>" buttons to toggle between regions. This option allows users to quickly jump between regions of interest, such as a list of known oncogene coordinates. 
<hr>

<a name="iz"></a>
### **Interactive Zooming**
As the **Zoom In** button increases the focuses in on the center part of the currently displayed, we describe how users can zoom on any region currently displayed. For example, consider the landscape of CTCF in 5 cell lines shown below. If a user is interested in further characterizing a particular peak, simply drag the mouse over the region of interest and define a shaded box. <br><br>
<img src="zoom1.png" width="50%" height="50%" />
<br><br>
After selecting a region, one can simply double click anywhere on the plot to zoom over the specified region. The boundaries of the shaded box will define the new x-coordinates in the updated plot. The y-coordinates of the box do not have an effect on the ensuing zoom. The plot below shows the shaded region after being double-clicked.
<br><br><br>
<img src="zoom2.png" width="50%" height="50%" /><br><br>
<hr>

<a name="uld"></a>
### **Visualizing Local Data**
While our browser provides a wide variety of epigenetic data for both mouse and human, users may wish to include their own data to visualize in the browser. **DNAlandscapeR** accepts the common data file formats (BedGraph, BigWig) for visualizing methylation, ChIP-Seq, RNA-Seq, and other quantiative regional data in addition to a new data format for ChIA-PET that we discuss [below](#crL) how to generate. The **Upload** tab contains the following input parameters to add local data to **DNAlandscapeR**:
* Specify organism-- Choose which genome build (human/mouse) to visualize the new data alongside.
* Add file-- Choose the local file to be uploaded to **DNAlandscapeR**. The selected file will not be imported into the browser until later input options are specified and the **Add** button is pressed. 
* Specify track name-- The text entry will define the name of the added track both in the plotting and in the dropdown selection.
* Data type-- Identify the file type (we do not infer the type from upload). ChIP-Seq, RNA-Seq, and other quantiative regional data should be specified as ReadDepth.
* **Add**-- The local file will hosted in browser alongside the other pre-loaded datasets. A user can confirm the configuration of the local addition in the data frame listed on the main panel of the page. 

<hr>
<a name="crL"></a>
### **Creating .rds for ChIA-PET Visualization**
Here, we walkthrough the creation of a new data format for visualizing DNA loops from ChIA-PET experiments. Specifically, the browser plots S4 `loops` objects that are produced from the R package [diffloop](https://bioconductor.org/packages/release/bioc/html/diffloop.html). As diffloop does not process the raw .fastq reads, we recommend using the [mango](https://github.com/dphansti/mango) preprocessing pipeline. Here, we provide a working example of how the K562-POL2cp track was generated from two sequencing runs [here](http://www.ncbi.nlm.nih.gov/sra?term=SRX107352) and [here](http://www.ncbi.nlm.nih.gov/sra?term=SRX107353).
<br>
First, one needs to download the .fastq files, which can achieved by installing the [SRA Toolkit](https://github.com/ncbi/sra-tools/wiki/HowTo:-Binary-Installation). Once this software is installed, the following commands will download the paired end read .fastq files--
```
fastq-dump --split-files SRR372747
fastq-dump --split-files SRR372748
```
Once these have downloaded, each paired end file was joined to make a single sample for mango to process.
```
cat SRR372747_1.fastq SRR372748_1.fastq > K562-POL2cp_1.fastq
cat SRR372747_2.fastq SRR372748_2.fastq > K562-POL2cp_2.fastq
```
The following command executes the mango ChIA-PET preprocessing pipeline once all dependencies are installed and each of the four files below are in the local directory. 
```
Rscript mango.R --fastq1 K562-POL2cp_1.fastq --fastq2 K562-POL2cp_2.fastq --prefix K562-POL2c_p --argsfile encode.mango.argsfile
```
where the specified .argsfile that corresponds to this particular sample follows-- <br>
`cat encode.mango.argsfile`
```
bowtieref         = ~/bowtie_index/hg19/hg19
bedtoolsgenome    = ~/bedtoolsgenome/human.hg19.genome
LinkerA           = GGCCGCGATATCGGATCCAAC
LinkerB           = GGCCGCGATATATGATCCAAC
chromexclude      = chrM,chrY
verboseoutput     = FALSE
reportallpairs    = TRUE
minPETS           = 1
```
On a high-performance cluster, this pre-processing step should take ~1 hour to complete. Several files are generated alongside the mango output, but we'll use the K562-POL2cp.interactions.all.mango, which should look something like this-- <br>

`head -3 K562-POL2cp.interactions.all.mango`
```
chr1	32705184	32708527	chr1	32712484	32715408	4	1
chr1	32705184	32708527	chr1	32755920	32759477	3	1
chr1	32705184	32708527	chr1	32799616	32803060	6	1
```
Add diffloop script and annotation files + awk commands
<hr>
<a name="hic"></a>
### **Creating .rds for HiC Visualization**
As HiC data, depending on the resolution, can vastly exceed the file sizes of other tracks, we compress the typical output of processed HiC data into a several matrices where each .rds contains one chromosome and the matrix values are the interaction scores for the particular chromosome. The coordinates of each chromosome are index in the row and column names. These matrices are saved/loaded in a .rds file for fast data access. We note that **DNAlandscapeR** utilizes only the lower triangle of the data matrix when plotting interaction scores. Below is a sample R script that was used to take output from [HiC-Pro](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0831-x) and convert the GM12878 HiC data at 1Mb resolution to an .rds object that **DNAlandscapeR** can plot. 
```
library(readr)
library(GenomicRanges)
library(reshape2)
library(tools)

matrix.file <- "gm12878_1000000_iced.matrix"
bed.file <- "gm12878_1000000_abs.bed"
out.pre <-  "gm12878_1000000"

bed.GRanges <- GRanges(data.frame(read_tsv(bed.file, col_names = c("chr", "start", "stop", "region"))))
dat.long <- read_tsv(matrix.file, col_names = c("idx1", "idx2", "region"))

core_chrom <- paste("chr", seq(1,22,1), sep = "")
core_chrom <- c(core_chrom, "chrX")

dir.create(out.pre)
lapply(core_chrom, function(chr){
    cur.chrom <- bed.GRanges[seqnames(bed.GRanges) == chr]
    vals <- mcols(cur.chrom)$region
    dat.chrom <- dat.long[dat.long$idx1 %in% vals & dat.long$idx2 %in% vals,]
    mat.chrom <- dcast(data = dat.chrom, formula = idx2 ~ idx1, value.var = "region", fill = NA)
    row.names <- as.character(mat.chrom[,1])
    mat.chrom <- mat.chrom[,-1]

    starts <- start(cur.chrom)
    names(starts) <- as.character(vals)

    rownames(mat.chrom) <- unname(starts[row.names])
    colnames(mat.chrom) <- unname(starts[colnames(mat.chrom)])
    saveRDS(as.matrix(mat.chrom), file = paste(out.pre, "/", out.pre, "-", chr, ".rds", sep=""))
})
```
Due to the prohibitive size of HiC data, the local upload supports adding only one chromosome of local data at a time. Each chromosome will then appear as its own track, which can cause problems if looking at different chromosomal regions in a session (e.g. displaying the chromosome 9 local HiC track while looking at chromosome 7). While the pre-loaded HiC data do not have this problem, users should proceed with caution when uploading local HiC data or contact <A HREF="mailto:caleblareau@g.harvard.edu">Caleb</A> to facilitate permanent file storage in the browser. 
<hr>
<a name="faq"></a>
### **FAQ**
#### **What are the build of the genomes in DNAlandscapeR?**
The current build of **DNAlandscapeR** is hg19 for human and mm9 for mouse. All epigenetic tracks and gene annotations pre-loaded in the browser utilize these coordinates.<br><br>

#### **How do I access the data pre-loaded into the browser?**
All data can be downloaded from our Amazon S3 instance using `wget`. For example, accessing the .bw of the HCT116  whole genome bisulfite sequencing data could be achieved by running the following command--
```
wget http://s3.amazonaws.com/dnalandscaper/data/human/methylation/HCT116-wgbs.bw
```
Each dataset is [indexed](http://s3.amazonaws.com/dnalandscaper) on the site. However, it may be easier to find the file by parsing the HTML to see the filenames. The following code performs this parsing in the `R` environment. Each track name in the browser had an identical file prefix indexed by the type of track that it displays.
```
amazon <- "http://s3.amazonaws.com/dnalandscaper"
xmlDat <- RCurl::getURL(amazon, ftp.use.epsv = FALSE, dirlistonly = TRUE)
amazon.filenames <- gsubfn::strapplyc(xmlDat, "<Contents><Key>(.*?)</Key>", simplify = c)
paste(amazon, amazon.filenames, sep = "/")
```
<br>

#### **How do I access the code used to make the plots/browser?**
**DNAlandscapeR** was built in the R/Shiny environment. All of our source code is made freely available in our [open-source repository](https://github.com/aryeelab/DNAlandscapeR).<br><br>

#### **I want to use a plot similar to what I made in DNAlandscapeR for a paper/presentation/etc., but I need greater control over the plot characteristics. What do I do?**
Our browser makes extensive use of the [Sushi](https://www.bioconductor.org/packages/release/bioc/html/Sushi.html) package in rendering images. We recommend that users use our source code and data in a local R environment alongside this library to generate more specialized plots. <br><br>

#### **Are the individual tracks normalized?**
Each epigenetic track is shown on it's own scale, which can be determined on the left side of the plot. Currently, loop widths are normalized between plots. Thus, the boldest loop that is displayed across all tracks is supported by the most PETs from that particular sequencing experiment. HiC data are normalized/displayed within each track, and the scale is noted on the right side of the track plot. <br><br>

#### **What do the colors mean in the loop plots?**
Consistent with [diffloop](https://bioconductor.org/packages/release/bioc/html/diffloop.html), blue loops are 'CTCF loops', purple loops are "Enhancer-Enhancer loops", orange loops are "Promoter-Promoter loops", red loops are 'Enhancer-Promoter loops', and black loops have no special annotation. Check out the [walkthrough](#crL) for and the diffloop vignette for a better description of how these loops are colored. Loops that fit multiple classifications were prioritzed based on the folowing hierarchy: E-P > P-P > E-E > CTCF > none. We note that black loops could still have biological annotation but were not identifed as using the current data processing. The Data Descriptions tab contains details on how individual samples were annotated and processed. Additionally, the mouse loops have not been specially annotated and thus are all black. <br><br>

#### **I'm tired of uploading my local data to visualize in the browser. Can I keep it in DNAlandscapeR permanently?**
Sure! We remind you that our browser is open source, so your data will thus be accessible to other users as well. If you want to add a track to **DNAlandscapeR**, contact <A HREF="mailto:caleblareau@g.harvard.edu">Caleb</A> to arrange for the inclusion of more data. We may have you process your data in a certain format to keep our data storage costs on Amazon low. 
